src/main/java/org/alliancegenome/curation_api/response/SupplementalDatahelper.java

```diff
@@ -0,0 +1,79 @@
+package org.alliancegenome.curation_api.response;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import io.quarkus.logging.Log;
+
+public class SupplementalDatahelper {
+
+	public static final String ERROR_MAP = "errorMap";
+	
+	public static void addFieldErrorMessages(Map<String, Object> supplementalData, String fieldName, Object fieldErrorMessages) {
+		Map<String, Object> errorMap = getErrorMap(supplementalData);
+		if(!errorMap.containsKey(fieldName)) {
+			errorMap.put(fieldName, fieldErrorMessages);
+		} else {
+			Log.warn("Error Map already contains errors for field: " + fieldName);
```

Comment:
@markquintontulloch this is the only question I have is this now happens globally as in if there is already a error message we check and throw this warning that it already exists.